contrib/endpoints/src/api_manager/weighted_selector.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+// includes should be ordered. This seems like a bug in clang-format?
+#include "contrib/endpoints/src/api_manager/weighted_selector.h"
+
+namespace google {
+namespace api_manager {
+
+WeightedSelector::WeightedSelector(
+    std::vector<std::pair<std::string, int>>&& list) {
+  list_.swap(list);
+  counts_.resize(list_.size());
+  std::fill(counts_.begin(), counts_.end(), 0);
+}
+
+float WeightedSelector::score(int index) {
+  return float(counts_[index]) / list_[index].second;
+}
+
+const std::string& WeightedSelector::Select() {
+  if (list_.size() == 1) {
```

Comment:
I don't think you need the == 1 check. The code should work even if you don't have the check. Rather you should check for an empty list instead here.